client/lib/track-form/index.jsx

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { union } from 'lodash';
+
+export const trackForm = WrappedComponent => {
+	class TrackFormComponent extends Component {
+		state = {
+			dirtyFields: [],
+			fields: {}
+		};
+
+		updateFields = ( fields, updateDirtyFields = true, callback ) => {
```

Comment:
I'm a little confused by this new second parameter. Would a separate `replaceFields` function be an appropriate alternative for the use-case of having passed `false` here?